src/main/java/fr/xephi/authme/service/BungeeService.java

```diff
@@ -1,69 +1,191 @@
             return;
         }
 
-        service.scheduleSyncDelayedTask(new Runnable() {
-            @Override
-            public void run() {
-                ByteArrayDataOutput out = ByteStreams.newDataOutput();
-                out.writeUTF("Connect");
-                out.writeUTF(bungeeServer);
-                player.sendPluginMessage(plugin, "BungeeCord", out.toByteArray());
-            }
-        }, 20L);
+        service.scheduleSyncDelayedTask(() ->
+            sendBungeecordMessage("Connect", player.getName(), destinationServerOnLogin), 20L);
+    }
+
+    private void sendAuthMeBungeecordMessage(String type, String... data) {
+        if(!isEnabled) {
+            return;
+        }
+
+        List<String> dataList = Arrays.asList(data);
+        dataList.add(0, "AuthMe");
+        dataList.add(1, type);
+        sendBungeecordMessage(dataList.toArray(new String[dataList.size()]));
+    }
+
+    public void sendLogin(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.LOGIN, name.toLowerCase());
+    }
+
+    public void sendLogout(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.LOGOUT, name.toLowerCase());
+    }
+
+    public void sendRegister(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REGISTER, name.toLowerCase());
+    }
+
+    public void sendUnregister(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.UNREGISTER, name.toLowerCase());
+    }
+
+    public void sendRefreshPassword(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_PASSWORD, name.toLowerCase());
+    }
+
+    public void sendRefreshSession(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_SESSION, name.toLowerCase());
+    }
+
+    public void sendRefreshQuitLoc(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_QUITLOC, name.toLowerCase());
+    }
+
+    public void sendRefreshEmail(String name) {
+        sendAuthMeBungeecordMessage(AuthMeBungeeMessageType.REFRESH_EMAIL, name.toLowerCase());
     }
 
     @Override
-    public void reload(Settings settings) {
-        this.isEnabled = settings.getProperty(HooksSettings.BUNGEECORD);
-        this.bungeeServer = settings.getProperty(HooksSettings.BUNGEECORD_SERVER);
-        Messenger messenger = plugin.getServer().getMessenger();
-        if (!this.isEnabled) {
+    public void onPluginMessageReceived(String channel, Player player, byte[] data) {
+        if(!isEnabled) {
             return;
         }
-        if (!messenger.isOutgoingChannelRegistered(plugin, "BungeeCord")) {
-            messenger.registerOutgoingPluginChannel(plugin, "BungeeCord");
+
+        ByteArrayDataInput in = ByteStreams.newDataInput(data);
+        String subchannel = in.readUTF();
+        if(!"Authme".equals(subchannel)) {
+            return;
+        }
+
+        String type = in.readUTF();
+        switch (type) {
```

Comment:
This switch doesn't handle all `AuthMeBungeeMessageType`.

Reply:
It's wanted, some messages are there for the bungee side plugin and future uses